raiden/tests/integration/long_running/test_token_networks.py

```diff
@@ -258,3 +258,54 @@ def test_participant_selection(raiden_network, token_addresses):
             channel_ids=channel_identifiers,
             retry_timeout=0.1,
         )
+
+
+@raise_on_failure
+@pytest.mark.parametrize("number_of_nodes", [2])
+@pytest.mark.parametrize("channels_per_node", [0])
+@pytest.mark.parametrize("settle_timeout", [10])
+@pytest.mark.parametrize("reveal_timeout", [3])
+def test_connect_does_not_open_channels_with_offline_nodes(raiden_network, token_addresses):
+    """
+    Test that using the connection manager to connect to a token network
+    does not open channels with offline nodes
+
+    Test for https://github.com/raiden-network/raiden/issues/5583
+    """
+    # pylint: disable=too-many-locals
+    registry_address = raiden_network[0].raiden.default_registry.address
+    token_address = token_addresses[0]
+    app0 = raiden_network[0]
+    app1 = raiden_network[1]
+    offline_node = raiden_network[0]
+    target_channels_num = 1
+
+    # connect the first node - this will register the token and open the first channel
+    # Since there is no other nodes available to connect to this call will do nothing more
+    RaidenAPI(app0.raiden).token_network_connect(
+        registry_address=registry_address,
+        token_address=token_address,
+        funds=TokenAmount(100),
+        initial_channel_target=target_channels_num,
+    )
+
+    # First node will now go offline
+    offline_node.raiden.stop()
+    offline_node.raiden.greenlet.get()
+    assert not offline_node.raiden
+
+    # Call the connect endpoint for app 1, the node which stayed online
+    RaidenAPI(app1.raiden).token_network_connect(
+        registry_address=registry_address,
+        token_address=token_address,
+        funds=TokenAmount(100),
+        initial_channel_target=target_channels_num,
+    )
+
+    # ensure that we did not open a channel with the offline node
```

Comment:
Could you add another node that get's connected to, to check that it actually connects to someone.

Reply:
the problem with adding another node was that they will connect to it and not connect to the offline one. But can't be sure that it's because the offline one is offline, or because the other node simply came first in the list.

So to simplify the test I left it like this. Since this test tests only that we don't connect to offline nodes. There is another test in the same file for connecting to online nodes.

Reply:
> the problem with adding another node was that they will connect to it and not connect to the offline one. But can't be sure that it's because the offline one is offline, or because the other node simply came first in the list.

I don't understand.
If we first connect node 0, then it will open a channel to 0x2222... . Then we cann connect node 1, which will connect to node 0. Then we can shut down node 0 and connect node2, which should then only connect to node 1.

Reply:
What I was trying to say is that I wrote the most simple test to make sure that the implemented change is tested. It succeeds with the changes in connection_manager and fails without them.

The scenario you are presenting is quite more complicated. But it actually reveals a flaw in the implementation!

> Then we cann connect node 1, which will connect to node 0. Then we can shut down node 0 and connect node2, which should then only connect to node 1.

When we shut down node 0, we shut down the only node that has made a channel in the token network. When the other 2 nodes try to connect to the token network there is nobody online and connected to connect to.

And they don't try to connect to the bootstrap address (`0x22222...`) because as you can see below 

https://github.com/raiden-network/raiden/blob/2b0f074215cb7f96b4ff70b377ae109c62f667d8/raiden/connection_manager.py#L165-L180

`qty_network_channels` is not zero since node 0 has made the first channel. So they don't connect to anybody.

So this is after all definitely worth a test.

But on how to solve it I have to think about it a bit.

Reply:
@palango I think the last commit I added in the PR: https://github.com/raiden-network/raiden/pull/5633/commits/fa9579ef2b5448e690fda6fba6774cb077e6d336 handled the case I mentioned above.

What do you think?